tests/tools/src/lib.rs

```diff
@@ -590,13 +590,16 @@ fn configure_command<'a>(
     script_result_directory: &Path,
 ) -> &'a mut std::process::Command {
     let never_path = if cfg!(windows) { "-" } else { ":" };
+    let mut msys_for_git_bash_on_windows = std::env::var("MSYS").unwrap_or_default();
+    msys_for_git_bash_on_windows.push_str(" winsymlinks:nativestrict");
```

Comment:
I decided to check to make sure this behaves correctly even when another `winsymlinks` [option](https://cygwin.com/cygwin-ug-net/using-cygwinenv.html) is already listed in `$MSYS`.

**This code is correct!** 😸 I say that based on this experiment:

```text
ek@Glub MINGW64 ~/tmp
$ touch target

ek@Glub MINGW64 ~/tmp
$ MSYS=winsymlinks:lnk ln -s target lnk-symlink

ek@Glub MINGW64 ~/tmp
$ MSYS=winsymlinks:nativestrict ln -s target nativestrict-symlink

ek@Glub MINGW64 ~/tmp
$ MSYS='winsymlinks:lnk winsymlinks:nativestrict' ln -s target lnk-nativestrict-symlink

ek@Glub MINGW64 ~/tmp
$ MSYS='winsymlinks:nativestrict winsymlinks:lnk' ln -s target nativestrict-lnk-symlink

ek@Glub MINGW64 ~/tmp
$ cmd //c dir
 Volume in drive C is OS
 Volume Serial Number is B203-10FB

 Directory of C:\Users\ek\tmp

07/06/2024  07:22 PM    <DIR>          .
07/06/2024  07:22 PM    <DIR>          ..
07/06/2024  07:21 PM    <SYMLINK>      lnk-nativestrict-symlink [target]
07/06/2024  07:21 PM               525 lnk-symlink.lnk
07/06/2024  07:22 PM               525 nativestrict-lnk-symlink.lnk
07/06/2024  07:21 PM    <SYMLINK>      nativestrict-symlink [target]
07/06/2024  07:19 PM                 0 target
               5 File(s)          1,050 bytes
               2 Dir(s)  157,514,899,456 bytes free
```

The later option overrides the earlier one, as one would expect. This doesn't seem to be documented for `CYGWIN` (in Cygwin) or `MSYS`, but I think it's reasonable to rely on. The alternative of removing prior `winsymlinks` options seems much less good to me, because that effect could be unintuitive when inspecting the environment and could also create the false impression that other options such as `error_start` would be removed, even though they would not be.

Reply:
Thanks a lot for trying it out! Just this morning I was thinking about this, wondering if it's working as expected :).